frontend/src/components/CustomTable.tsx

```diff
@@ -189,6 +189,8 @@ interface CustomTableProps {
   getExpandComponent?: (index: number) => React.ReactNode;
   initialSortColumn?: string;
   initialSortOrder?: 'asc' | 'desc';
+  initialFilterString?: string;
+  setFilterString?: (filterString: string) => void;
```

Comment:
These parameters now allow developers to set an initial filter string, and save the current filter string in the parent component. If maintaining this filter string after run re-selection is not important and not worth these added parameters, this can be removed; otherwise, I will add this logic as shown.